docs/Maintainer-Guidelines.md

```diff
@@ -15,6 +15,28 @@ All Homebrew maintainers are encouraged to contribute to all parts of the projec
 
 These documents are meant to serve as guiding principles. As a maintainer, you can make a call to either request changes from a contributor or help them out based on their comfort and previous contributions. Remember, as a team we [Prioritise Maintainers Over Users](Maintainers-Avoiding-Burnout.md) to avoid burnout. If you wish to change or discuss any of the guidelines: open a PR to suggest a change.
 
+## Reviewing PRs
+
+When reviewing a PR, use "approve", "approve with comments", "comment" or "request changes" when submitting based on the following guidelines:
+
+- ✅ Approve: if you feel that the PR looks good as is, you can ✅ approve it as-is.
+- ✅ Approve with comments: if you have a few questions or comments to be answered and the PR can be merged after they are addressed, you can ✅ approve with comments.
+  - Please trust that other maintainers will not merge the PR until the comments are addressed rather than e.g. making them wait another 24h to get another review.
+  - If auto-merge is enabled: you can disable it (add `automerge-skip` on homebrew-core/cask) if you want to ensure the PR is not merged until your comments are at least read.
+- 🗣️ Comment: if you need to ask questions before you can provide a ✅ approval but are fine with someone else providing a ✅ approval before you, provide a comment review and ask questions.
+- 🚫 Request changes: a last resort. Save it for "if this PR is merged before I personally have had a chance to ✅ approve it: it seems very likely to cause user-visible problems".
```

Comment:
Does this also apply to core? Because while there are no user-visible problems from user binaries there I feel my usage of this change is appropriate for them.

Reply:
Also for the cask repos. I think that using "changes requests" indicates clearly to the PR that the PR cannot be merged until the changes are made. Other maintainers may dismiss the review if the changes have been made before an additional pass.

Reply:
@SMillerDev @bevanjkay Yes, this would apply everywhere (and we should have a consistent policy/treatment everywhere generally).

While it's possible to dismiss a "request changes": it feels culturally more friendly and saves everyone time and effort to save this for the case where "a specific person needs to rereview this PR before it should be merged".

Reply:
Do we have another way to indicate that something is fundamentally wrong and the user should redo it?

Reply:
As stated here: "request changes" should be more "it's not worth making code changes until we've discussed if we even want this PR".

In general: people don't like "request changes". It is perceived as negative/aggressive and, in your case, it can be avoided by just not commenting and instead stating what you would like to be changed.

Reply:
I certainly have never considered a request changes review as being negative, but a way of expressing that the changes are required before the PR can go ahead. 

I do find it very confusing that an approving review can have comments that require addressing before merging.

This conflicts somewhat with GitHub's own [documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#about-pull-request-reviews) about the feature, so it feels a bit confusing to me. 

> A review has three possible statuses:

> Comment: Submit general feedback without explicitly approving the changes or requesting additional changes.
> Approve: Submit feedback and approve merging the changes proposed in the pull request.
> Request changes: Submit feedback that must be addressed before the pull request can be merged.

Reply:
> I certainly have never considered a request changes review as being negative, but a way of expressing that the changes are required before the PR can go ahead.

Sure, but I've had others reach out who do feel this way.

> I do find it very confusing that an approving review can have comments that require addressing before merging.

GitHub does not provide a "apply these suggestions and then you can merge".

The main issue here is that we regularly see cases where a PR is blocked for ~24 hours (longer over weekends) because someone has commented/requested changes but not ✅ so the PR _cannot be merged_ until after that's been done.

> This conflicts somewhat with GitHub's own [documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews#about-pull-request-reviews) about the feature, so it feels a bit strange to me.

Their documentation is written for all of their customers. I can state from my 10 years experience of working there (on many timezone-distributed teams): my documentation above better matches how GitHub uses this feature than their documentation does 😅

Reply:
I can understand where the delay can cause issues for sure.

In the past, at least in `homebrew-cask`, we have used a label to indicate that a response is required from the PR author or feedback required. It has been more recently that there has been a general movement toward "request changes". For non-trivial changes some PRs would bounce between being labelled "awaiting user reply" and "awaiting maintainer feedback", for visibility purposes.

Reply:
@bevanjkay Ok, thanks for explaining.

Would you feel better about this if we stated to only avoid using "request changes" like this when reviewing work from other maintainers?

Reply:
I'm ok with the changes after hearing your reasoning, it creates less maintainer friction for merging the changes after the PR looks good. 

I would say in contrast to the Brew repo, that most code review suggestions in `homebrew-cask` are blockers. 

As an aside, we've shifted culturally to providing suggestions and coaching PRs to create better contributors, where in the past maintainers would just commit the changes. 

Thanks for discussing 👍🏻

Reply:
Throwing my bit in here, I echo @bevanjkay's comment.  I have used it in Cask for "this should be changed prior to approval" and typically are blockers, not suggestions.  I use comments for those.  And never considered it to be a requirement that the person requesting changes be the reviewing party.

I always took it at face value of "please make this change to meet [best practice | guideline]" so never considered to be bothered by it. 😃 

No argument - just noting where my thoughts are and have differed.  Happy to approve.